src/XAD/JITGraphInterpreter.hpp

```diff
@@ -0,0 +1,487 @@
+            }
+            case JITOpCode::Remquo:
+            {
+                int quo;
+                XAD_UNUSED_VARIABLE(std::remquo(va, vb, &quo));
+                nodeAdjoints_[a] += adj;
+                nodeAdjoints_[b] -= adj * quo;
+                break;
+            }
+            case JITOpCode::Hypot:
+                nodeAdjoints_[a] += adj * va / vResult;
+                nodeAdjoints_[b] += adj * vb / vResult;
+                break;
+            case JITOpCode::Nextafter:
+                nodeAdjoints_[a] += adj;
+                // Second operand has zero derivative
+                break;
+            case JITOpCode::Ldexp:
+            {
+                double imm = graph.immediates[nodeId];
+                int exp = static_cast<int>(imm);
+                nodeAdjoints_[a] += adj * (1 << exp);
+                break;
+            }
+            case JITOpCode::Frexp:
+            {
+                // Derivative is 1 / 2^exp, but we need to recompute frexp
+                int exp;
+                std::frexp(va, &exp);
+                nodeAdjoints_[a] += adj / (1 << exp);
+                break;
+            }
+            case JITOpCode::Modf:
+                // Derivative of fractional part is 1
+                nodeAdjoints_[a] += adj;
+                break;
+            case JITOpCode::Copysign:
+                // d/da copysign(a, b) = sign(b)
+                nodeAdjoints_[a] += adj * ((vb >= 0.0) ? 1.0 : -1.0);
+                // d/db copysign(a, b) = 0
+                break;
+            case JITOpCode::SmoothAbs:
+            {
+                double dval;
+                if (va > vb)
+                    dval = 1.0;
+                else if (va < -vb)
+                    dval = -1.0;
+                else if (va < 0.0)
+                    dval = va / (vb * vb) * (3.0 * va + 4.0 * vb);
+                else
+                    dval = -va / (vb * vb) * (3.0 * va - 4.0 * vb);
+                nodeAdjoints_[a] += adj * dval;
+
+                // Derivative w.r.t. c (second parameter)
+                double dcval;
+                if (va > vb || va < -vb)
+                    dcval = 0.0;
+                else if (va < 0.0)
+                    dcval = -2.0 * va * va * (vb + va) / (vb * vb * vb);
+                else
+                    dcval = -2.0 * va * va * (vb - va) / (vb * vb * vb);
+                nodeAdjoints_[b] += adj * dcval;
+                break;
+            }
+            case JITOpCode::CmpLT:
+            case JITOpCode::CmpLE:
+            case JITOpCode::CmpGT:
+            case JITOpCode::CmpGE:
+            case JITOpCode::CmpEQ:
+            case JITOpCode::CmpNE:
+                break;
+            case JITOpCode::If:
+                if (va != 0.0)
+                    nodeAdjoints_[b] += adj;
+                else
+                    nodeAdjoints_[c] += adj;
+                break;
+            default:
+                break;
+        }
+    }
+};
```

Comment:
There is a lot of code in this class and we would like to see more granular unit tests of this. Basically only this class, trying different inputs that eventually exhaust all code paths, including error cases, and checking if the output is as expected.